src/components/general/Typography/Typography.tsx

```diff
@@ -16,14 +16,34 @@ export const Typography = (props: ITypographyProps) => (
   </ConfigProvider>
 )
 
+type TypographySize = 'base' | 'sm' | 'lg' | 'xl'
 export interface ITextProps extends AntTextProps {
+  size?: TypographySize
+}
+
+const getFontSizeVariable = (size: TypographySize): string => `--font-size${size === 'base' ? '' : `-${size}`}`
+
+const getLineHeightVariable = (size: TypographySize): string => {
+  if (size === 'sm') return `--line-height-sm`
+  if (size === 'lg' || size === 'xl') return '--line-height-lg'
+  return `--line-height`
+}
+
+const Text = ({ size = 'base', ...props }: ITextProps) => {
+  return (
+    <ConfigProvider>
+      <AntTypography.Text
+        style={{
+          fontSize: `var(${getFontSizeVariable(size)})`,
+          lineHeight: `var(${getLineHeightVariable(size)})`,
+        }}
+        {...props}>
+        {props.children}
+      </AntTypography.Text>
+    </ConfigProvider>
```

Comment:
An alternative to inline styling would've been nesting a config provider

```suggestion
    <ConfigProvider>
      <AntTypography.Text
        style={{
          fontSize: `var(${getFontSizeVariable(size)})`,
          lineHeight: `var(${getLineHeightVariable(size)})`,
        }}
        {...props}>
        {props.children}
      </AntTypography.Text>
    </ConfigProvider>
```